generators/entity-client/files.js

```diff
@@ -301,16 +315,18 @@ function writeFiles() {
                     false,
                     this.fetchFromInstalledJHipster(`entity-client/templates/${CLIENT_NG2_TEMPLATES_DIR}`)
                 );
-                this.addEntityToModule(
-                    this.entityInstance,
-                    this.entityClass,
-                    this.entityAngularName,
-                    this.entityFolderName,
-                    this.entityFileName,
-                    this.entityUrl,
-                    this.clientFramework,
-                    this.microserviceName
-                );
+                if (!this.embedded) {
```

Comment:
Became too confusing, must be refactored. 

```
let variables***;
let microserviceName = this.microserviceName;
if (this.clientFramework === 'angularX') {
    variables = somethingAngular
} else if (this.clientFramework === 'react') {
    variables = somethingReact;
    microserviceName = undefined;
} else {
    this.addEntityToMenu(this.entityStateName, this.enableTranslation, this.clientFramework, this.entityTranslationKeyMenu);
    return;
}
this.writeFilesToDisk( variables );
addEnumerationFiles(this, CLIENT_NG2_TEMPLATES_DIR, ANGULAR_DIR);
if (!this.isEmbeded) {
    this.addEntityToModule(variables, microserviceName);
    this.addEntityToMenu(variables);
}
```

***variables should include every specific variable like:
clientFiles(writeFilesToDisk), clientTemplatesDir(writeFilesToDisk), enumTemplatesDir(addEnumerationFiles), enumDestDir(addEnumerationFiles)

I think you should create a different PR for this refactoring, I can review it.

Reply:
Just pushed the a PR with the refactoring https://github.com/jhipster/generator-jhipster/pull/11300.
Once merged, just rebase and add the necessary condition.

Reply:
The refactoring PR is merged, thanks @mshima!